activerecord/test/cases/migration/check_constraint_test.rb

```diff
@@ -123,6 +123,21 @@ def test_add_check_constraint
           end
         end
 
+        def test_add_check_constraint_with_if_not_exists_options
+          @connection.add_check_constraint :trades, "quantity > 0"
+
+          assert_nothing_raised do
+            @connection.add_check_constraint :trades, "quantity > 0", if_not_exists: true
+          end
+
+          # SQLite never raises an error anyway
+          unless current_adapter?(:SQLite3Adapter)
+            assert_raises(ActiveRecord::StatementInvalid) do
+              @connection.add_check_constraint :trades, "quantity > 0"
+            end
+          end
```

Comment:
I am not sure this part is needed here. 1) We test for `if_not_exists` option in this test case and 2) this behavior is kinda obvious to need a test?